IBX-8599: DAM integration with DXP (#2476)
ibexa/documentation-developer: docs/dam/integration_with_dxp.md

---
description: Integrate DAM with your Ibexa DXP project to use an effective tool for managing marketing assets.
---

# Integrate DAM with DXP

DAM integration with [[= product_name =]] is dedicated to [[= product_name_base =]] users who also use Actito services. It enables using assets repository stored in the DXP in the Actito platform to build and insert marketing assets into campaign emails.

> For more information, see [Actito documentation](https://cdn3.actito.com/fe/actito-documentation/docs/Using_Assets_from_a_DAM/#connection-to-ibexa).

Reviewer: I'm not sure about this. Integration with Actito has dedicated instance ibexa/dam. We can integrate DAM with regular DXP instance every time when we want to use DXP as data storage. Integration with Actito is irrelevant here. Correct me if I'm wrong @dkorostynski.

## Enable Elasticsearch or Solr

To ensure proper indexing and search results, DAM requires one of
the following search engines on your server:

- [Elasticsearch](install_elastic_search.md)
- [Solr](install_solr.md)

> ## Enable OAuth Servers

Reviewer: ```suggestion
## Enable OAuth Server
```

Next, enable 0Auth Server authentication. To do it, follow steps up to the [Client section](oauth_server.md).

## Install DAM package

Install dedicated DAM `ibexa/dam-user` package into your [[= product_name =]] instance.

To install, run the following command:

```bash
composer require ibexa/dam-user
```

Add the `ibexa_dam_oauth2_client_user` table to the database:

```bash
php bin/console ibexa:doctrine:schema:dump-sql vendor/ibexa/dam-user/src/bundle/Resources/config/schema.yaml | mysql -u <username> -p <database_name>
```

> Next, add technical users, required by DAM to your [[= product_name =]] instance.

Reviewer: It is worth to mention about two technical users that will be created

- `image-picker-reader` - with read permission 
- `image-picker-editor` - with edit / delete permission 

Both users have limited access to Media folder only.

To do it, first, import the migration file:

```bash

>     php bin/console ibexa:migrations:import vendor/ibexa/dam-user/src/bundle/Resources/migrations/2024_07_11_09_10_dam_technical_users.yaml

Reviewer: ```suggestion
php bin/console ibexa:migrations:import vendor/ibexa/dam-user/src/bundle/Resources/migrations/2024_07_11_09_10_dam_technical_users.yaml
```

```

Next, run migrations:

```bash
php bin/console ibexa:migrations:migrate
```

Two new user roles and users should be now visibile in the Back Office.

Create OAuth2 client for the DXP user:

```bash

> php bin/console ibexa:dam:create-oauth2-client <client-name> <user-identifier>

Reviewer: Will be good to mention about using technical user identifiers (image-picker-reader, image-picker-editor)

```

## Override admin-ui

> Add the following configuration in the `yaml` file, for example, `ibexa_admin_ui_settings.yaml`:

Reviewer: Maybe some explanation here, why we need to override these parameters. These parameters are added for admin_group and we need to override all of them and use site_group scope to make them available under REST API.

        ibexa.site_access.config.site_group.subtree_operations.copy_subtree.limit: 100

        # Notifications
        ibexa.site_access.config.site_group.notifications.error.timeout: 0
        ibexa.site_access.config.site_group.notifications.warning.timeout: 0
        ibexa.site_access.config.site_group.notifications.success.timeout: 5000
        ibexa.site_access.config.site_group.notifications.info.timeout: 0
    ```

## Add CROSS-ORIGIN config

To integrate DAM you need to add in the `config/packages/nelmio_cors.yaml` following configuration:

```yaml
nelmio_cors:
    defaults:
        origin_regex: true
        allow_origin: ['%env(CORS_ALLOW_ORIGIN)%']
        allow_methods: ['GET', 'OPTIONS', 'POST', 'PUT', 'PATCH', 'DELETE', 'PUBLISH', 'COPY']
        allow_headers: ['Content-Type', 'Authorization', 'X-HTTP-Method-Override']
        expose_headers: ['Link']
        max_age: 3600
    paths:
        '^/api/ibexa/v2/':
            allow_methods: ['GET', 'OPTIONS', 'POST', 'PUT', 'PATCH', 'DELETE', 'PUBLISH', 'COPY']
            allow_headers: ['Content-Type', 'Authorization', 'X-HTTP-Method-Override']
```

Next, in the `.env` file provide path with a regular expression:

> `CORS_ALLOW_ORIGIN='^https?://(localhost|127\.0\.0\.1|(.+\.acme\.(com|eu)))?(:[0-9]+)?$'`

Reviewer: Maybe some remark about changing acme domain to your own.

---
description: Integrate DAM with your Ibexa DXP project to use an effective tool for managing marketing assets.
---

# Integrate DAM with DXP

DAM integration with [[= product_name =]] is dedicated to [[= product_name_base =]] users who want to use [[= product_name_base =]] as a data storage. It enables using assets repository stored in the DXP and manage assets. 

## Enable Elasticsearch or Solr

To ensure proper indexing and search results, DAM requires one of
the following search engines on your server:

- [Elasticsearch](install_elastic_search.md)
- [Solr](install_solr.md)

## Enable OAuth Server

> Next, enable 0Auth Server authentication. To do it, follow steps up to the [Client section](oauth_server.md).

Reviewer: ```suggestion
Next, enable OAuth Server authentication. To do it, follow steps up to the [Client section](oauth_server.md).
```

---
description: Integrate DAM with your Ibexa DXP project to use an effective tool for managing marketing assets.
---

# Integrate DAM with DXP

> DAM integration with [[= product_name =]] is dedicated to [[= product_name_base =]] users who want to use [[= product_name_base =]] as a data storage. It enables using assets repository stored in the DXP in the platform to build and insert marketing assets into campaign emails.

Reviewer: This sentence "to build and insert marketing assets into campaign emails." still "smells" of Actito integration.

> For more information, see [Actito documentation](https://cdn3.actito.com/fe/actito-documentation/docs/Using_Assets_from_a_DAM/#connection-to-ibexa).

Reviewer: ```suggestion
```

## Enable Elasticsearch or Solr

To ensure proper indexing and search results, DAM requires one of
the following search engines on your server:

- [Elasticsearch](install_elastic_search.md)
- [Solr](install_solr.md)

## Enable OAuth Server

Next, enable 0Auth Server authentication. To do it, follow steps up to the [Client section](oauth_server.md).

## Integration with Platform.sh

If you have your [[= product_name =]] project deplyed on Platform.sh, the following steps are required:

> In the `.env` file, set paths to OAuth Server.

Reviewer: ```suggestion
In the `.platform.app.yaml` file, set paths for private and public keys to OAuth Server.
```

Next, generate these keys and add them.

```bash

> Platform.sh
> 
> .platform.app.yaml
> 

Reviewer: ```suggestion
```
